flink-runtime/src/main/java/org/apache/flink/runtime/io/network/buffer/Buffer.java

```diff
@@ -227,4 +229,53 @@
 	 * Tags the buffer as compressed or uncompressed.
 	 */
 	void setCompressed(boolean isCompressed);
+
+	/**
+	 * Gets the type of data this buffer contains.
+	 */
+	DataType getDataType();
+
+	/**
+	 * Used to identify the type of data contained in {@link Buffer}.
+	 */
+	enum DataType {
+		/**
+		 * DATA_BUFFER indicates that this buffer represents a non-event data buffer.
+		 */
+		DATA_BUFFER(true, false),
+		/**
+		 * EXACTLY_ONCE_CHECKPOINT_BARRIER indicates that this buffer represents a
+		 * serialized checkpoint barrier of exactly once checkpoint mode.
+		 */
+		EXACTLY_ONCE_CHECKPOINT_BARRIER(false, true),
+		/**
+		 * UNKNOWN_EVENT indicates this buffer represents serialized data of other
+		 * unknown event.
+		 */
+		UNKNOWN_EVENT(false, false);
+
+		private final boolean isBuffer;
+
+		private final boolean isExactlyOnceCheckpointBarrier;
```

Comment:
It does not need to maintain this so specific property, otherwise when we further extend the data type enum future, it also needs to extend the respective field here.

`isBuffer` is a general property to be retained for widely use.  Then we can also judge via `getDataType == DataType.EXACTLY_ONCE_CHECKPOINT_BARRIER`  to replace `isExactlyOnceCheckpointBarrier`.